packages/core/src/tracing/cls.js

```diff
@@ -256,18 +272,21 @@ class InstanaIgnoredSpan extends InstanaSpan {
  * @param {string} [spanAttributes.traceId]
  * @param {string} [spanAttributes.parentSpanId]
  * @param {import('./w3c_trace_context/W3cTraceContext')} [spanAttributes.w3cTraceContext]
+ * @param {Object} [spanAttributes.spanContextData]
  * @returns {InstanaSpan}
  */
 
 function startSpan(spanAttributes = {}) {
-  let { spanName, kind, traceId, parentSpanId, w3cTraceContext } = spanAttributes;
+  let { spanName, kind, traceId, parentSpanId, w3cTraceContext, spanContextData } = spanAttributes;
 
   tracingMetrics.incrementOpened();
   if (!kind || (kind !== ENTRY && kind !== EXIT && kind !== INTERMEDIATE)) {
     logger.warn(`Invalid span (${spanName}) without kind/with invalid kind: ${kind}, assuming EXIT.`);
     kind = EXIT;
   }
-  const span = new InstanaSpan(spanName);
+  const spanData = spanContextData || {};
+
+  const span = new InstanaSpan(spanName, spanData);
```

Comment:
suggestion: Forward spanContextData straight into the class. Rename spanContextData to spanData in function argument